Adding 2018 single track reconstruction efficiency utility/params

chughes90 opened the pull request:
Adding in parameters for 2018 single track reconstruction efficiency + utilities to header (.h) and .cxx. Parameters are from fits to LHC18l8a4 HIJING simulations. Should work similar to 2015 and 2011 functionalities in previous version of code.

alibuild commented:
af88ce2d8f2f9c8f48155e45c77cbd2764fd960d: approval required: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @matplo (Mateusz Ploskon), @davidrohr (David Rohr), @adriansev (Adrian Sevcenco), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan), @mfasDa (Markus Fasel), @lcunquei (Leticia Cunqueiro Mendez), @qgp (Jochen Klein)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/19487) and comment at the bottom of the page.**_

chughes90 commented:
Modified Files:

AliAnalysisTaskEmcalJetHUtils.cxx
AliAnalysisTaskEmcalJetHUtils.h

Description:

Adds parameters/functionality for 2018 single track reconstruction efficiency.

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetHUtils.cxx:
```diff
@@ -244,7 +284,8 @@ void AliAnalysisTaskEmcalJetHUtils::ConfigureEventCuts(AliEventCuts & eventCuts,
                             std::make_pair("LHC15o", &AliEventCuts::SetupLHC15o),
-                            std::make_pair("LHC18qr", &AliEventCuts::SetupPbPb2018) };
+                            std::make_pair("LHC18qr", &AliEventCuts::SetupPbPb2018),
+                            std::make_pair("LHC18q", &AliEventCuts::SetupPbPb2018) };
```
If you need LHC18q separately, I would imagine that you would want LHC18r separately here (I'm not sure you would need it separately, but it's also fine)

chughes90 replied:
Added in flag for LHC18r

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetHUtils.cxx:
```diff
@@ -2,6 +2,7 @@
 // Utilities class for Jet-Hadron correlation analysis
 //
 
+//#include "AliAnalysisTaskEmcalJetHUtils.h"
```
spurious comment

chughes90 replied:
Removed

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetHUtils.h:
```diff
@@ -2,7 +2,7 @@
 
 /**
- * @class AliAnalysisTaskEmcalJetHUtils
+ * @class AliAnalysisTaskEmcalJetH	Utils
```
spurious spaces

chughes90 replied:
Removed

raymondEhlers commented on PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetHUtils.h:
```diff
@@ -126,12 +129,30 @@ class AliAnalysisTaskEmcalJetHUtils {
  * All functions in the FlowVectorCorrections static class are copied diretly from the AddTask. I wish I didn't have
  * to do this, but I need it in compiled code, so I have no choice.
  */
+/*
```
I guess you don't need this functionality, which is fine. If it's not preventing the code from compiling, you should remove the comment

chughes90 replied:
Removed everything below line 150

raymondEhlers reviewed (changes requested):
Code looks good - just a few small comments, which should be quick + easy to fix

alibuild commented:
6f6fae83eab803781a4602f84c2bbb37641aa38e: approval required: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @matplo (Mateusz Ploskon), @davidrohr (David Rohr), @adriansev (Adrian Sevcenco), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan), @mfasDa (Markus Fasel), @germainma (Marie Germain), @lcunquei (Leticia Cunqueiro Mendez), @qgp (Jochen Klein)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/19487) and comment at the bottom of the page.**_

chughes90 commented:
Pushed a new commit with all of Raymond's feedback incorporated:

Modified Files:

AliAnalysisTaskEmcalJetHUtils.cxx
AliAnalysisTaskEmcalJetHUtils.h

Description:

Added flag for LHC18r and removed all spurious comments/spaces

alibuild commented:
5ae6c0e6d4659f75060453cf1757f5c1722ef517: approval required: 1 of @jgrosseo (Jan Fiete Grosse-Oetringhaus), @matplo (Mateusz Ploskon), @davidrohr (David Rohr), @adriansev (Adrian Sevcenco), @raymondEhlers (Raymond James Ehlers), @loizides (Constantinos Loizides), @vkucera (Vit Kucera), @shahor02 (Ruben Shahoyan), @mfasDa (Markus Fasel), @germainma (Marie Germain), @lcunquei (Leticia Cunqueiro Mendez), @qgp (Jochen Klein)

_Comment with `+1` to approve and allow automatic merging,or with `+test` to run tests only. **Please comment on the pull request: [click here](https://github.com/alisw/AliPhysics/pull/19487) and comment at the bottom of the page.**_

chughes90 commented:
Pushed a new commit correcting issues with the Flow Vector Corrections class in .h

Modified Files:

AliAnalysisTaskEmcalJetHUtils.h

Description:

Added back in class declaration for the Flow Vector Corrections to .h

raymondEhlers commented:
+1

alibuild commented:
5ae6c0e6d4659f75060453cf1757f5c1722ef517: approved: will be automatically merged on successful tests
